### PR TITLE
app/cody: small window UI for cody chat page

### PR DESCRIPTION
**Pull request by limitedmage**
Closes https://github.com/sourcegraph/sourcegraph/issues/53112

Significantly improves the small window/mobile UI in the Cody chat page, perfect for docking the App on the side of the screen.

## Test plan

https://github.com/sourcegraph/sourcegraph/assets/206864/406589c6-a73b-41a8-8f97-86e6f7c34021

### CODE REVIEW

**Review by marekweb — APPROVED** [review]
Nice!

**Comment by almeidapaulooliveira** [comment]
Nice work. A couple of observations:

- Can we make the breakpoint early? I mean, the sidebar it's too narrow before switching to another view.
- On the 'All chats' view, we're missing the 'Start a new chat' button and the 'Clear all chats' button. I was expecting behavior similar to what we have in the sidebar on the web app.

File: client/web/src/cody/chat/CodyChatPage.module.scss [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
     filter: none;
     border: 1px solid var(--border-color-2);
     border-radius: var(--border-radius);
     padding: 0;
 }
 
+.page {
+    @media (--sm-breakpoint-down) {
+        // These need to be !important to override the `py-4` default padding on the page
+        padding-top: 0 !important;
```
Comment (vovakulikov) [inline client/web/src/cody/chat/CodyChatPage.module.scss:13]: Note for myself, don't set paddings on root components with spacing classNames

File: client/web/src/cody/chat/CodyChatPage.tsx [hunk 2]
```diff
@@ -285,9 +303,54 @@ export const CodyChatPage: React.FunctionComponent<CodyChatPageProps> = ({ authe
                         ))}
                 </div>
 
-                <div className={classNames('d-flex flex-column col-sm-9 h-100', styles.chatMainWrapper)}>
+                <div
+                    className={classNames(
+                        'd-md-flex flex-column col-md-9 h-100',
```
Comment (vovakulikov) [inline client/web/src/cody/chat/CodyChatPage.tsx:308]: It isn't a blocker. Mixing spacing/blocks classnames and custom module classname styles may make it harder to read, understand and debug. I suggest having these styles in one place, if it's something simple classnames are just fine but if we had to create a designated `module.scss` file this means that we need to put all styles there for a better readability.

**Review by vovakulikov — APPROVED** [review]
Nice mobile view!

File: client/web/src/cody/chat/CodyChatPage.tsx (hunk 2 above)
Comment (limitedmage) [reply client/web/src/cody/chat/CodyChatPage.tsx:308]: Good call. I've moved all the conditional mobile styles to the scss file.

**Pull request merged**